app/Domains/Vendors/Actions/UpdateVendor.php

```diff
@@ -25,9 +25,14 @@ public function authorize(ActionRequest $request): Response
         return Response::deny('You are unauthorised to perform this action');
     }
 
-    public function handle(Vendor $vendor, array $params): Vendor
+    public function handle(string $id, array $params): Vendor
     {
-        $vendor->update($params);
+        $vendor = Vendor::findOrFail($id);
+        $vendor->name = $params['name'] ?? $vendor->name;
```

Comment:
We can do a normal `$vendor->update($params)` here. There are no sensitive data here that needs to be address individually.

Reply:
Issues resolved